dask_cuda/explicit_comms/dataframe/shuffle.py

```diff
@@ -165,24 +194,76 @@ def multi_shuffle_group(
     ret: Dict[int, DataFrame] = {}
     for i in range(npartitions):  # Iterate over all possible output partition IDs
         t = [df_grouped[i] for df_grouped in dfs_grouped]
+        assert len(t) > 0
         if len(t) == 1:
-            ret[i] = t[0]
+            ret[i] = proxify(t[0])
         elif len(t) > 1:
             ret[i] = proxify(dd_concat(t, ignore_index=ignore_index))
-        else:
-            ret[i] = df_meta  # Empty dataframe
```

Comment:
Can you explain why we don't need to cover this case anymore? Is it just because we now have no entry at all for an empty DataFrame? Note that I'm happy that we are no longer passing around the _meta if we don't need to be.

Also, is the comment above still true? ("If the partition is empty, an empty dataframe is used.")

Reply:
`group_split_dispatch()` and thus `partition_dataframe()` will always return a dataframe per partition even if some of them becomes empty. 
I cannot remember why the meta was needed in the first place. I have added tests of the different batchsizes and everything seems to work: 268e63fcbb8b6719b8401b02439e3c92834a3a30

Reply:
>group_split_dispatch() and thus partition_dataframe() will always return a dataframe per partition even if some of them becomes empty.

Ah - Sorry. I didn't see the `assert len(t) > 0` line above this. It makes sense to remove this if the zero-length case is explicitly forbidden anyway.